### PR TITLE
feat(optimus): Add clear button to date time field

### PR DESCRIPTION
**Pull request by Gladkowski**
#### Summary

Add clear button to `input filed` and `date time` field.

- [input filed design ](https://mews.design/7b4bb574e/p/5037e4-text-input/b/32e1a2) 
- date time field is a mobile specific component, and probably we should move it to commander client

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

File: storybook/pubspec.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  intl: ^0.17.0
```
Comment (chaeMil) [inline storybook/pubspec.yaml:12]: Why introducing this library? And dragging this to the end user.

File: storybook/lib/stories/date_time_field.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  Widget build(BuildContext context) => ConstrainedBox(
+        constraints: const BoxConstraints(maxWidth: 400),
+        child: OptimusDateTimeField(
+          value: _dateTime,
+          isClearAllEnabled:
+              widget.k.boolean(label: 'Clear all', initial: false),
+          label: widget.k.text(label: 'Label', initial: 'Optimus input field'),
+          error: widget.k.text(label: 'Error', initial: ''),
+          placeholder: widget.k.text(label: 'Placeholder', initial: ''),
+          formatDateTime: DateFormat().format,
```
Comment (chaeMil) [inline storybook/lib/stories/date_time_field.dart:37]: This is not necessary, just provide a function.

**Review by chaeMil — CHANGES_REQUESTED** [review]

**Review by chaeMil — APPROVED** [review]

File: optimus/lib/src/date_time_field.dart [hunk 3]
```diff
@@ -6,24 +6,28 @@ import 'package:optimus/optimus.dart';
...
+      {Key? key,
+      this.value,
+      this.label,
+      required this.onChanged,
+      this.minDate,
+      this.maxDate,
+      this.error,
+      required this.formatDateTime,
+      this.isClearAllEnabled = false,
+      this.placeholder})
```
Comment (ookami-kb) [inline optimus/lib/src/date_time_field.dart:19]: Add trailing comma.

File: optimus/lib/src/date_time_field.dart [hunk 4]
```diff
@@ -6,24 +6,28 @@ import 'package:optimus/optimus.dart';
...
 
   final DateTime? value;
   final String? label;
-  final ValueChanged<DateTime> onChanged;
+  final ValueChanged<DateTime?> onChanged;
   final DateTime? minDate;
   final DateTime? maxDate;
   final DateTimeFormatter formatDateTime;
   final String? error;
+  final bool isClearAllEnabled;
```
Comment (ookami-kb) [inline optimus/lib/src/date_time_field.dart:29]: I would rather rename it to `isClearEnabled` everywhere, all doesn't add any value.

File: optimus/lib/src/input_field.dart [hunk 5]
```diff
@@ -123,6 +129,21 @@ class _OptimusInputFieldState extends State<OptimusInputField>
...
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            _ClearAllButton(onTap: () {
+              _controller?.clear();
+              if (widget.onChanged != null) widget.onChanged!('');
+            }),
+            if (widget.suffix != null) widget.suffix!,
+          ],
+        )
+      : widget.suffix;
```
Comment (ookami-kb) [inline optimus/lib/src/input_field.dart:145]: Can be simplified:

```suggestion
  Widget get _suffix => OptimusStack(
        direction: Axis.horizontal,
        spacing: OptimusStackSpacing.spacing100,
        mainAxisSize: MainAxisSize.min,
        children: [
          if (widget.isClearAllEnabled)
            _ClearAllButton(
              onTap: () {
                _controller?.clear();
                widget.onChanged?.call('');
              },
            ),
          if (widget.suffix != null) widget.suffix!,
        ],
      );
```

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: optimus/lib/src/input_field.dart [hunk 6]
```diff
@@ -114,6 +119,7 @@ class _OptimusInputFieldState extends State<OptimusInputField>
   void initState() {
     super.initState();
     _effectiveFocusNode.addListener(_onFocusChanged);
+    _controller = widget.controller ?? TextEditingController();
```
Comment (ookami-kb) [inline optimus/lib/src/input_field.dart:122]: If you're creating `TextEditingController` you should also dispose of it, it's the same as with focus nodes.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by Gladkowski** [comment]
@ookami-kb ping

File: optimus/lib/src/input_field.dart [hunk 7]
```diff
@@ -106,6 +110,10 @@ class _OptimusInputFieldState extends State<OptimusInputField>
     with ThemeGetter {
   FocusNode? _focusNode;
   bool _isShowPasswordEnabled = false;
+  TextEditingController? _controller;
+
+  TextEditingController get _effectiveController =>
+      widget.controller ?? _controller!;
```
Comment (ookami-kb) [inline optimus/lib/src/input_field.dart:116]: Why `!`? Use the same construction as in the focus node.

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**